File: CPlusPlus/169. 多数元素.cpp
/*
执行用时：12 ms, 在所有 C++ 提交中击败了91.79% 的用户
内存消耗：19.1 MB, 在所有 C++ 提交中击败了50.47% 的用户
*/
class Solution {
public:
    int majorityElement(vector<int>& nums) {
        int count = 0;
        int result;
        for(int i = 0; i < nums.size(); ++i){
            if(count == 0) {
                count++;
                result = nums[i];
            }
            else if(nums[i] == result) ++count;
            else{
                --count;
            }
        }
        return result;
    }
};

File: Rust/169. 多数元素.rs
impl Solution {
    pub fn majority_element(nums: Vec<i32>) -> i32 {
        let mut count:i32 = 0;
        let mut current:i32 = 0;
        for elem in nums{
            if count == 0 {
                current = elem;
                count = 1;
            }else if elem == current{
                count = count + 1;
            }else{
                count = count - 1;
            }
        }
        current
    }
}